app/controllers/backgrounds_controller.rb

```diff
@@ -26,15 +30,27 @@ def edit
   # POST /backgrounds
   # POST /backgrounds.json
   def create
-    @background = Background.new(background_params)
+    name = params[:name]
+    description = params[:description]
+    sample_ids = params[:sample_ids]
 
-    respond_to do |format|
+    if sample_ids.any? { |sid| !current_power.samples.pluck(:id).include?(sid) }
```

Comment:
I think you can simplify this by doing sample_ids.to_set - current_power.samples.pluck(:id).to_set and make sure it's empty.